crates/runtime/src/datafusion.rs

```diff
@@ -183,7 +184,10 @@ pub enum Table {
 pub struct DataFusion {
     pub ctx: Arc<SessionContext>,
     data_writers: RwLock<HashSet<TableReference>>,
-    pub cache_provider: RwLock<Option<Arc<QueryResultsCacheProvider>>>,
+    cache_provider: RwLock<Option<Arc<QueryResultsCacheProvider>>>,
+
+    /// Has the initial load of the data been completed? It is the responsibility of the caller to call `mark_initial_load_complete` when the initial load is complete.
+    initial_load_complete: Mutex<bool>,
```

Comment:
I think this is good enough to start with. But later we might need to introduce more complex state tracking.